### ISSUE TITLE
Unable to store an Observable

### ISSUE THREAD

**Issue by andre-lima** [DanWahlin/Observable-Store#61]
Hello Dan,

I'm trying to store some observables instead of the object itself so I can make use of the async pipe at the end of the data flow.

Although, I'm getting this error
core.js:6210 ERROR TypeError: Converting circular structure to JSON

I guess the cloning logic is not compatible with an Observable.

Would that be the issue? What workaround is possible while maintaining the Observables?

Thanks!

**Comment by DanWahlin** [comment]
Hi Andre. Store state needs to contain something that can be serialized at that point in time so storing an observable in the store isn't possible. An observable is a stream of data into the future so there's no way to capture all of its "state" upfront since more data can come later.

Having said that, you can still use observables all the way from the service to the view. One example from the samples can be found here (https://github.com/DanWahlin/Observable-Store/blob/master/samples/angular-store/src/app/orders/orders.component.ts) but there are many other ways observables could be used so that async pipe can be applied in a view. A service will likely expose a function or property that interacts with the store and returns an observable. Or, a component may access the `stateChanged` property (https://github.com/DanWahlin/Observable-Store#store-api) exposed by a service that extends `ObservableStore` and then use the async pipe to get to the data that changed.

Bottom line is the store doesn't support storing an observable in it. Only state that can be serialized can be stored. However, observables themselves can be used all the way from services to components to views.
